src/slices/GiveThunk.ts

```diff
@@ -53,10 +52,15 @@ export const changeApproval = createAsyncThunk(
     }
 
     const signer = provider.getSigner();
-    const mockSohmContract = new ethers.Contract(addresses[networkID].MOCK_SOHM as string, MockSohm, signer);
+    let sohmContract;
+    if (networkID === 1) {
+      sohmContract = new ethers.Contract(addresses[networkID].SOHM_ADDRESS as string, ierc20Abi, signer);
+    } else if (networkID === 4) {
+      sohmContract = new ethers.Contract(addresses[networkID].MOCK_SOHM as string, MockSohm, signer);
```

Comment:
Document why this requires a different contract, since it's a different file.

Reply:
will do